packages/domain/lib/usecase/get_movie_anticipated_response_usecase.dart

```diff
@@ -0,0 +1,40 @@
+import '../utils/const.dart';
+import '../model/movie_anticipated_response.dart';
+import '../repository/movie_repository.dart';
+import 'usecase.dart';
+
+class GetMovieAnticipatedResponseUseCase
+    implements UseCase<Future<List<MovieAnticipatedResponse>>> {
+  final NetworkRepository _repository;
+
+  GetMovieAnticipatedResponseUseCase(this._repository);
+
+  @override
+  Future<List<MovieAnticipatedResponse>> call() async {
+    final List<MovieAnticipatedResponse> jsonMovies = [];
+    final response =
+        await _repository.getMovieData(apiPath: C.apiAnticipatedPath);
+    if (int.parse(response.headers['x-pagination-page-count'][0]) >= 5) {
```

Comment:
1) move header name to constants
2) move parsing pageCount value to separate variable - it will make debugging easier in future